Component/add evaluation form

PReichetanz opened the pull request:
Refactor evaluations: added now by choice of rating and phrase

![Bildschirmfoto 2021-12-07 um 16 31 08](https://user-images.githubusercontent.com/33567019/145058410-24a1cff5-48fa-4779-b73b-1b86f800f0f0.png)

- Add button hover and active colour
- Add defaultEvaluations in categories and 5 marks
- Add basic useEvaluation.ts for localStorage usage
- In PupilOverview, you can add a evaluation by choosing the category, rating and a prompted phrase.


Hint: I focused on the AddEvaluationForm component; when adding a pupil, you can still add a category and evaluation by writing - this will be removed in the next PR.

[View at heroku](https://capstone-rev-component--kgk8zr.herokuapp.com/)

PReichetanz committed:
Add colour for button hover and active

PReichetanz committed:
Implement basic AddEvaluationForm component +story

PReichetanz committed:
Remove unnecessary onClick function in type

PReichetanz committed:
Add map and styling of evaluation options

PReichetanz committed:
Add DefaultEvaluations.ts

- Content: evaluations for four categories and 5 marks, respectively

PReichetanz committed:
Add useEvaluation with adjusted typest.ts

PReichetanz committed:
Add display of evaluations dependent on rating

PReichetanz committed:
Add setName function for proper evaluations

- setName replaces X in DefaultEvaluations.ts with first name of pupil

PReichetanz committed:
Fix missing setName call in isActive prop

F-Kirchhoff commented on src/app/components/AddEvaluationForm/AddEvaluationForm.tsx:
```diff
@@ -0,0 +1,171 @@
+  onCancel,
+}: AddEvaluationFormProps): JSX.Element {
+  const { evaluations } = useEvaluations();
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [selectedEvaluation, setSelectedEvaluation] = useState('');
+  const [rating, setRating] = useState(0);
```
looks a bit inconsistent, maybe remove selected from the first two states or rename rating to selectedRating

F-Kirchhoff commented on src/app/components/AddEvaluationForm/AddEvaluationForm.tsx:
```diff
@@ -0,0 +1,171 @@
+    if (selectedCategory === '' || rating === 0 || selectedEvaluation === '') {
+      setInputError(true);
+      return;
+    } else {
+      const category = selectedCategory;
+      const evaluation = selectedEvaluation;
+      onSubmit(category, evaluation);
+      setInputError(false);
+      setSelectedCategory('');
+      setSelectedEvaluation('');
+    }
+  }
```
when you return in your if statement, then you dont have to wrap the rest into an else statement.

F-Kirchhoff commented on src/app/components/AddEvaluationForm/AddEvaluationForm.tsx:
```diff
@@ -0,0 +1,171 @@
+          />
+        ))}
+
+        {inputError && selectedCategory === '' && (
+          <SubmitWarning>Bitte geben Sie eine Kategorie ein.</SubmitWarning>
+        )}
```
nice way of solving this problem

F-Kirchhoff commented on src/app/components/AddEvaluationForm/AddEvaluationForm.tsx:
```diff
@@ -0,0 +1,171 @@
+        {selectedCategory &&
+          evaluations.map((evaluation) =>
+            evaluation.name === selectedCategory
+              ? evaluation.evaluations.map((evaluation) =>
+                  evaluation.mark === rating
+                    ? evaluation.descriptions.map((description, key) => (
```
there is a evaluations array inside evaluation inside evaluations? this is very confusing

F-Kirchhoff reviewed (approved):
good work!

PReichetanz committed:
Add changes according to review

- rename defaultEvaluations.evaluation to valuations
- Update references to valuations
- Rename rating to selectedRating
